trinity/events.py

```diff
@@ -0,0 +1,13 @@
+from lahja import (
+    BaseEvent
+)
+
+
+class PeerCountRequest(BaseEvent):
+
+    def __init__(self) -> None:
+        super().__init__(None)
```

Comment:
I think that `BaseEvent` should be updated so that it doesn't require this hack.  I'd suggest using a *sentinal* empty object to allow distinguishing between `None` and no-payload-provided.

```python
class EmptyPayload:
    pass

empty_payload = EmptyPayload()


class BaseEvent:
    def __init__(self, payload=empty_payload):
        ...
```

*or*

It may just make sense to have payload only accessible via a method like `get_payload()` and raise an exception if no payload was provided so that events with no payload cannot accidentally have their payload used for something when it is empty.

Honestly it's not clear what the right thing to do here is wrt payloads but I think this hack should be removed.

Reply:
I was even considering getting rid of `payload` entirely and let the events specify whatever fields they need.